Sources/Core/Microsoft.StreamProcessing.Provider/Extensions/Linq.cs

```diff
@@ -0,0 +1,76 @@
+﻿// *********************************************************************
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License
+// *********************************************************************
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Microsoft.StreamProcessing.Provider
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public struct Empty
+    { }
+
```

Comment:
Can we make this private?

Reply:
It's temporary. We should decide whether the Empty type lives in the engine or the provider DLL and put it in only one place. Alternatively, we may not need it at all in the provider DLL if we do something different with the SelectMany API.